### ISSUE TITLE
Booster shot with BioNTech on J&J shows "wait X days until completed" text

### ISSUE THREAD

**Issue by reckter** [corona-warn-app/cwa-app-ios#3814]
## Avoid duplicates
* [x] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [ ] Bug is specific for iOS only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
 => I can not test on Android, so idk, happy to move this issue, when necessary.
* [x] Bug is not already reported in another issue

## Technical details

- Device name: iPhone 12 pro
- iOS version: 15.1
- App version: 2.13.1 (0)

## Describe the bug

After importing a booster-shot certificate for BioNTech, while already having a J&J certificate imported the app shows the text in the picture below on the certificate screen. All other information on the screen still correctly reflect the first successful vaccination though. It also states that it only uses the J&J certificate as the main one. 

![IMG_967E9B456B34-1](https://user-images.githubusercontent.com/1771450/140944454-eae295da-7d09-4883-bd9a-b51beee5694e.jpeg)

## Steps to reproduce the issue

Upload the booster certificate no later then 14 days after the vaccination.

## Expected behaviour

I would have expected to not see the text. 
I might misinterpret the text, but as it's only the booster vaccination my certificate should be valid in these 14 days as well, which is not reflected in the text. 

## Possible Fix

Probably the easiest fix is to not display the text, when the certification without the last vaccination shot is sufficient.

**Comment by Ein-Tim** [comment]
This problem is known and can't be fixed, see https://github.com/Digitaler-Impfnachweis/certification-apis/discussions/198.

**Comment by reckter** [comment]
@Ein-Tim Ah thank you.
But can't you distinguish between BioNTech on J&J and biontech "normal" vaccination, by the numbers?
Shouldn't biotech be 2/2 after 1/2? instead of 2/2 on 1/1? Or am I mistaken there?

**Comment by Ein-Tim** [comment]
@reckter 

Each certificate has to be "looked at" independently of any other certificates, because the EU rules only allow "one _event_ in one certificate".

The problem wouldn't be to hide the message you are seeing for everyone who got a Biontech "2/2" certificate and a Jansen & Jansen "1/1" certificate, this would just be looking at the certificates you've stored in your app and hiding this message.
However, this wouldn't make your "2/2" certificate valid, so it makes no sense to do this. 

Hope this was clear?

**Comment by reckter** [comment]
> The workaround for users with a 1/1-J&J vaccination is to present for checking in the first 14 days the J&J certificate and the 2/2 booster certificate.

As far as I can tell, this is what the CWA, currently does as well, correct?
I still stand by my point coming from the user perspective:
Effectively the app is telling me that my vaccination is not completed yet, when in fact it is.
I think hiding the text, and choosing the correct certificate to show in the QR code, dependent on the date, is the most user friendly approach.

> The problem wouldn't be to hide the message you are seeing for everyone who got a Biontech "2/2" certificate and a Jansen & Jansen "1/1" certificate, this would just be looking at the certificates you've stored in your app and hiding this message.
> However, this wouldn't make your "2/2" certificate valid, so it makes no sense to do this.

My underlying problem is, that I do have valid vaccination coverage, but the app is suggesting me that I currently do not have that.
I understand the technical reasons now (thank for explaining, much appreciated!), even though I think the message should be altered to not confuse users who do not know these reasons.

Explaining why the new certificate will only be shown in X days is my preferred solution currently ("Your newest certificate is valid in X days, until then the app will show your previous, still valid, certificate" or so).

All that aside, this is not an app braking bug or anything.

**Comment by ndegendogo** [comment]
> Each certificate has to be "looked at" independently of any other certificates, because the EU rules only allow "one event in one certificate".

> In principle one could include a logic onto the CWA, can can recognise a 2/2 BioNTech/Moderna certificate as a booster vaccination, if CWA finds a 1/1 J&J certificate in the "wallet". As pointed out by @Ein-Tim, the issue would still persist, since a CheckApp (in Germany CovPassCheck-App) only scans one certificate, it cannot conclude if it is a booster vaccination or the completion of a series of vaccination (Grundimmunisierung).

@Ein-Tim @dsarkar I don't know how this check is implemented in the app; but as a user, I would expect the following:
- check each certificate independently against the downloaded business rules *(I guess the question could remain, which business rules, as each country has its own rules ...)*
- don't "invent" cwa-specific additional rules - this could make the logic inconsistent and fragile (and, after all, the "master" is CovPassCheck)
- If there are several certificates **for the same person**, combine their tesults. If at least one is valid, the overall status for that person is valid.

*Does this make sense?*

**Comment by Ein-Tim** [comment]
@ndegendogo 

AFAIK this is how it's done since version 2.13.2. Do you have a way to confirm this?

@reckter For the reasons mentioned above this issue can't be fixed on CWA side, thus I think it doesn't really make sense to leave this issue open any longer, or? 

@dsarkar I suggest to apply the `won't fix` label here or create a new `can't fix` label and apply it here.

**Comment by ndegendogo** [comment]
> AFAIK this is how it's done since version 2.13.2. Do you have a way to confirm this?

@Ein-Tim I guess reading the code ... maybe another day 🙃

**Comment by moabits** [comment]
I don't think a "wontfix" is a solution here, at least the text in the app should be changed to include this information.
Something like in the CovPass FAQ: https://twitter.com/achisto/status/1467516734907293710

**Comment by dsarkar** [comment]
@moabits see also https://www.coronawarn.app/de/faq/#vac_booster_jj. @Ein-Tim do you want to create a wishlist item, that such information should be shown in the app, or maybe even better, a link to this (or another) FAQ article in the vaccination status box?

**Comment by reckter** [comment]
I agree with @moabits. If there isn"t a ticket in the right place, to get this fixed it should be created, and we can use this as a tracking for that. (I whole heartedly agree though, that as a maintainer of the cwa such tickets are frustrating, because you can't actively fix them)

> AFAIK this is how it's done since version 2.13.2. Do you have a way to confirm this?

if that's the case, then we can close this as fixed :D 

Looking at these lines: It seems like it's at least fixed in the Compass app: https://github.com/Digitaler-Impfnachweis/covpass-ios/blob/67278c30862e601131c01503331e3cf04bf3776b/Source/CovPassCommon/Sources/CovPassCommon/Models/Vaccination.swift#L32
But I have no idea about that code, or the inner workings of cwa and covpass, so I can't pass a final judgement. (and of course I can't test it either ^^)

**Comment by moabits** [comment]
Press reports: https://bnn.de/karlsruhe/karlsruhe-stadt/fehler-apps-digitales-impfzertifikat-booster-johnson-impfungen-genesene-corona-warnapp-covass-apotheken

**Comment by svengabr** [comment]
Not creating a new Jira ticket at this point but this one is getting added to our ongoing hot-topic discussion that is already active in Jira. The team is aware of this issue and actively trying to find a solution for this issue.

EXPOSUREAPP-10993
EXPOSUREAPP-11043

The EU has currently declined a 3 of 3 vaccination approach to fix this issue. There is a discussion going on how a "Germany-Wide" solution could be implemented in the meantime.

We have already published a faq article here:
https://www.coronawarn.app/de/faq/#vac_booster_jj

**Comment by svengabr** [comment]
Internal Tracking ID: EXPOSUREAPP-10993 + EXPOSUREAPP-11043
